odbc/src/api/functions.rs

```diff
@@ -2761,9 +2804,49 @@ unsafe fn set_connect_attrw_helper(
     let sql_return = {
         let conn = must_be_valid!((*conn_handle).as_connection());
 
+        let has_connection = conn.has_mongo_connection();
+
+        // Attributes could be set before or after the mongo_connection is established, so we
+        // have to check and write to the proper place. Keep in mind the conn.attributes Option<RwLock<ConnectionAttributes>> is TAKEN
+        // when a mongo_connection is established, so writing to both isn't safe
         match attribute {
             ConnectionAttribute::SQL_ATTR_LOGIN_TIMEOUT => {
-                conn.attributes.write().unwrap().login_timeout = Some(value_ptr as u32);
+                if has_connection {
+                    conn.mongo_connection
+                        .write()
+                        .unwrap()
+                        .as_mut()
+                        .unwrap()
+                        .attributes
+                        .login_timeout = Some(value_ptr as u32);
+                } else {
+                    conn.attributes
+                        .write()
+                        .unwrap()
+                        .as_mut()
+                        .unwrap()
+                        .login_timeout = Some(value_ptr as u32);
+                }
+                SqlReturn::SUCCESS
+            }
+            ConnectionAttribute::SQL_ATTR_CURRENT_CATALOG => {
+                let db = input_text_to_string_w(value_ptr as *const WideChar, str_length as usize);
+                if has_connection {
+                    conn.mongo_connection
+                        .write()
+                        .unwrap()
+                        .as_mut()
+                        .unwrap()
+                        .attributes
+                        .current_catalog = Some(db);
+                } else {
+                    conn.attributes
+                        .write()
+                        .unwrap()
+                        .as_mut()
+                        .unwrap()
+                        .current_catalog = Some(db);
```

Comment:
Thinking about ripping this out into a function.

Reply:
Actually, this would be a bit harder without a wrapper to generalize it.

Reply:
It's fine.